pageserver/src/page_service.rs

```diff
@@ -942,6 +950,13 @@ where
     ) -> Result<(), QueryError> {
         let ctx = self.connection_ctx.attached_child();
         debug!("process query {query_string:?}");
+        fail::fail_point!(
+            "simulated-bad-compute-connection",
```

Comment:
Does returning an error from here result in a disconnection?  Looking at `PostgresBackend::process_message`, it seems like errors from `process_query` just send an error response and proceed to continue servicing requests on the connection.

Reply:
Good point, I was checking the logs initially and I saw that the failpoint was getting hit, but now looking at it again it seems that it's only the first failpoint that gets hit. Once a connection is established it never goes away. I'll need to find another way

Reply:
See my other comment to Joonas about the configurable timeout strategy now.